packages/fela/index.d.ts

```diff
@@ -1,13 +1,44 @@
-export interface Renderer {
-    clear(): void;
-    renderRule(rule: Function, props?: any): string;
-    renderKeyframe(keyframe: any, props?: any): string;
-    renderFont(family: string, files: any[], properties?: any): string;
-    renderStatic(style: string | any, selector?: string): string;
+declare module "fela" {
+
+  import React from 'react';
+
+  type TRuleProps = {};
+  type TRule = (props: TRuleProps) => IStyle; 
+  type TKeyFrame = TRule;
+  type TRendererCreator = (config?: IConfig) => IRenderer;
+  type TPlugin = (style: IStyle) => IStyle; //http://fela.js.org/docs/advanced/Plugins.html
+  type TEnhancer = (renderer: IRenderer) => IRenderer; //http://fela.js.org/docs/advanced/Enhancers.html
+
+  interface IRenderer {
+    renderRule(rule: TRule, props: TRuleProps): void;
+    renderKeyframe(keyFrame: TKeyFrame, props: TRuleProps): void;
+    renderFont(family: string, files: Array<string>, props: TRuleProps): void;
+    renderStatic(style: string, selector?: string): void;
     renderToString(): string;
-    subscribe(callback: Function): any;
+    subscribe(event: (msg: string) => void): { unsubscribe: () => void; }
```

Comment:
Gotcha, Seems like the docs are out-dated kinda.

Consider the following cases:

1: Rules & Static style (object)
```javascript
// type === 1
{ selector: string, declaration: string, media: string, static?: boolean, type: number }
```

2. Static Style (string)
```javascript
// type === 4
{ type number, css: string }
```

3. Keyframes



export const RULE_TYPE = 1
export const KEYFRAME_TYPE = 2
export const FONT_TYPE = 3
export const STATIC_TYPE = 4
export const CLEAR_TYPE = 5